src/provider/cloud-element-templates/Condition.js

```diff
@@ -30,7 +30,17 @@ function isConditionMet(element, properties, property) {
 function isSimpleConditionMet(element, properties, condition) {
   const { property, equals, oneOf } = condition;
 
-  const propertyValue = getValue(element, properties, property);
+  let propertyValue;
+
+  if (element) {
+
+    // applying template to element
+    propertyValue = getValue(element, properties, property);
+  } else {
+
+    // creating element with template: use default value
+    propertyValue = getDefaultValue(properties, property);
```

Comment:
The one issue that I see with this approach is that we account for properties (and their value) that are conditionally deactivated (won't otherwise be applied). We'd need to solve that "conflict" one way or the other.

A sane alternative to what you implemented would be:

* Define a clear processing order => Properties are established first to last
* For each property in definition order check if it is valid against the currently established binding values => If yes, establish binding, if not ignore.

The benefit of that approach is that we could move the logic out of this utility, to the [template factory](https://github.com/bpmn-io/bpmn-js-properties-panel/pull/878/files#diff-3e86785fc01cbc4b7bd468798cec81083d46dfcffa00f780d25500b850b0c4c7). This utility could remain as is. It simply checks the condition against currently bound values available in the element.

Reply:
You sketch a third approach in https://github.com/bpmn-io/internal-docs/issues/718#issuecomment-1423938684: Mirror the existing update behavior.

(1) first apply all properties
(2) then hide all properties that no longer apply (per configuration, executed from top to bottom (first to last property))

Reply:
> To achieve the existing behaviour means changing the [create helpers](https://github.com/bpmn-io/bpmn-js-properties-panel/tree/master/src/provider/cloud-element-templates/create) to use the modeling.updateProperties API

Not necessarily. We would not _re-use_ but simply ensure that what is being executed (and how filtering happens) works in the exact same way:

* First we setup the element (one step, all properties)
* Then we remove properties that are not valid

Reply:
My bad, I did not get that right about pre-filtering.

> What you are describing would mean implementing this part (bellow) again?

I'm not suggesting any implementation. I'm just suggesting that we align both behaviors and have _one_ defined behavior. If that is already the case / in place, that is great. 

If it is not in place yet we want to get there, either sharing the algorithm _or_ at implementing it in the same spirit. As creating an element from a template is not invoking the `CommandStack` sharing won't be easily possible, I think.

Reply:
With https://github.com/bpmn-io/bpmn-js-properties-panel/pull/878#discussion_r1104591164 I was referring to the update behavior, where we first set, and then remove properties.

Seems conditional behavior is a little bit all over the place :).

Reply:
We further discussed this issue and found that the algorithm for safe creation of properties boils down to (1) create a dependency tree (2) establish properties in order of that tree, going from no dependencies -> dependent on most things:

![property-binding excalidraw](https://user-images.githubusercontent.com/58601/218771297-9a0b436a-5156-4579-95dd-7e23a291a1e1.png)